source/OctoVersion.Core/OctoVersionRunner.cs

```diff
@@ -52,13 +52,16 @@ public void Run(out OctoVersionInfo versionInfo)
             if (!string.IsNullOrWhiteSpace(appSettings.FullSemVer))
             {
                 Log.Information("Adopting previously-provided version information {FullSemVer}. Not calculating a new version number.", appSettings.FullSemVer);
-                var semanticVersion = SemanticVersion.TryParse(appSettings.FullSemVer);
+                var semanticVersion = SemanticVersion.TryParse(appSettings.FullSemVer!);
                 if (semanticVersion == null) throw new Exception("Failed to parse semantic version string");
 
                 versionInfo = new OctoVersionInfo(semanticVersion);
             }
             else
             {
+                //this should never happen - the AppSettings Validator makes sure that either appSettings.FullSemVer or appSettings.CurrentBranch is set 
+                if (string.IsNullOrEmpty(appSettings.CurrentBranch))
```

Comment:
Technically I shouldn't need this, but I couldn't figure out [the incantation](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/attributes/nullable-analysis) I needed